src/xmake.ts

```diff
@@ -256,6 +256,19 @@ export class XMake implements vscode.Disposable {
         this._projectFileSystemWatcher.onDidCreate(this.onProjectFileUpdated.bind(this));
         this._projectFileSystemWatcher.onDidChange(this.onProjectFileUpdated.bind(this));
 
+        const compileCommandsFile = path.join(config.compileCommandsDirectory, "compile_commands.json");
+        const compileCommandsWatcher = vscode.workspace.createFileSystemWatcher(compileCommandsFile);
+        compileCommandsWatcher.onDidCreate(() => {
+            const config = vscode.workspace.getConfiguration('C_Cpp');
```

Comment:
If the user does not have cpptools installed, do we always get a valid config instead of undefined?